### PR TITLE
[#327] Enable oauth2 on nexus & grafana ingress endpoint

### PR DESCRIPTION
**Pull request by MaksymBilenko**
This PR is related to #327 for Nexus

### CODE REVIEW

**Comment by MaksymBilenko** [comment]
ok to test

**Comment by kirillmakhonin** [comment]
This PR does not authorize users in nexus (does not give access to management). It only closes nexus from anonymous access.

**Comment by kirillmakhonin** [comment]
```
TASK [legion_core_chart : Pre run with dumping] ********************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "helm --kube-context legion-ci.epm.kharlamov.biz install legion-core --name legion-core --debug --dry-run -f /tmp//legion-core-values.legion-ci.epm.kharlamov.biz.yaml", "delta": "0:00:00.469630", "end": "2018-08-29 17:48:42.189115", "msg": "non-zero return code", "rc": 1, "start": "2018-08-29 17:48:41.719485", "stderr": "Error: failed to parse /tmp//legion-core-values.legion-ci.epm.kharlamov.biz.yaml: error converting YAML to JSON: yaml: line 73: did not find expected key", "stderr_lines": ["Error: failed to parse /tmp//legion-core-values.legion-ci.epm.kharlamov.biz.yaml: error converting YAML to JSON: yaml: line 73: did not find expected key"], "stdout": "[debug] Created tunnel using local port: '45111'\n\n[debug] SERVER: \"127.0.0.1:45111\"\n\n[debug] Original chart version: \"\"\n[debug] CHART PATH: /var/lib/jenkins/workspace/Deploy_Legion_Release/deploy/helms/legion-core", "stdout_lines": ["[debug] Created tunnel using local port: '45111'", "", "[debug] SERVER: \"127.0.0.1:45111\"", "", "[debug] Original chart version: \"\"", "[debug] CHART PATH: /var/lib/jenkins/workspace/Deploy_Legion_Release/deploy/helms/legion-core"]}
	to retry, use: --limit @/var/lib/jenkins/workspace/Deploy_Legion_Release/deploy/ansible/deploy-legion.retry
```

**Comment by ablatov** [comment]
@alexander-semenets do you have any ideas about the error above?

**Comment by alexander-semenets** [comment]
looks like a syntax error in legion-core values template

**Comment by MaksymBilenko** [comment]
ok to test

**Comment by MaksymBilenko** [comment]
ok to test

**Comment by aliaksandr-d** [comment]
`TASK [legion_enclave : Attach Airflow S3 accesse policy to the role] ***********
fatal: [localhost]: FAILED! => {"changed": true, "msg": "Rate exceeded"}`

**Comment by ablatov** [comment]
Checked manually on the cluster. Nexus was secured, Robot tests passed for this part. 
@MaksymBilenko in which task we will secure Grafana(not enclave one) too?

**Comment by MaksymBilenko** [comment]
@ablatov I'll add it here

**Comment by MaksymBilenko** [comment]
ok to test

**Comment by MaksymBilenko** [comment]
@akharlamov And what is your suggestion? For me that's the best way to go at this moment

**Comment by MaksymBilenko** [comment]
ok to test

**Comment by akharlamov** [comment]
Maybe deploy/helms/legion-core/templates/nexus/nexus-service.yaml ?

**Comment by akharlamov** [comment]
Approved given time sensitive nature of delivery.

**Comment by MaksymBilenko** [comment]
@akharlamov Please comment desired state or create an issue for this one.
I'm not clear what do you see wrong with ansible values template and how do you see this done?...

**Comment by MaksymBilenko** [comment]
@ablatov please fix failing test

**Comment by kirillmakhonin** [comment]
ok to test

**Comment by aliaksandr-d** [comment]
Check Nexus availability
```

  | Element 'xpath: //span[contains(@class, 'x-tree-node-text') and contains(text(), "Components")]' not visible after 30 seconds.
-- | --
```

**Comment by aliaksandr-d** [comment]
ok to test

**Comment by ablatov** [comment]
ok to test

File: deploy/ansible/roles/legion_core_chart/templates/legion-core-values.yaml.j2 [hunk 1]
```diff
@@ -79,12 +79,24 @@ grafana:
     tag: "{{ legion_version }}"
 
   ingress:
+{% if dex.enabled == true %}
```
Comment (akharlamov) [inline deploy/ansible/roles/legion_core_chart/templates/legion-core-values.yaml.j2]: The conditional logic should not be inside values file. Ingress templates is a more proper place for that.

**Comment by ablatov** [comment]
bot, test all

**Comment by ablatov** [comment]
bot, test all

**Comment by ablatov** [comment]
bot, test app

**Comment by ablatov** [comment]
bot, test all

**Comment by ablatov** [comment]
bot, test all

**Comment by aliaksandr-d** [comment]
bot, test apps

**Comment by MaksymBilenko** [comment]
bot, test app

**Comment by dsuslov** [comment]
it was done in PR #705

**Pull request closed without merging**